pkg/reconciler/kubernetes/tektonhub/tektonhub.go

```diff
@@ -0,0 +1,595 @@
+	kubeClientSet kubernetes.Interface
+	// operatorClientSet allows us to configure operator objects
+	operatorClientSet clientset.Interface
+	// manifest is empty, but with a valid client and logger. all
+	// manifests are immutable, and any created during reconcile are
+	// expected to be appended to this one, obviating the passing of
+	// client & logger
+	manifest mf.Manifest
+	// Platform-specific behavior to affect the transform
+	extension common.Extension
+	// enqueueAfter enqueues a obj after a duration
+	enqueueAfter func(obj interface{}, after time.Duration)
+}
+
+var (
+	apiConfigMapName string = "api"
+	errKeyMissing    error  = fmt.Errorf("secret doesn't contains all the keys")
+	namespace        string = ""
+	// Check that our Reconciler implements controller.Reconciler
+	_ tektonhubconciler.Interface = (*Reconciler)(nil)
+	_ tektonhubconciler.Finalizer = (*Reconciler)(nil)
+)
+
+const (
+	dbInstallerSet          = "DbInstallerSet"
+	dbMigrationInstallerSet = "DbMigrationInstallerSet"
+	apiInstallerSet         = "ApiInstallerSet"
+
+	releaseVersionKey  = "operator.tekton.dev/release-version"
+	createdByKey       = "operator.tekton.dev/created-by"
+	targetNamespaceKey = "operator.tekton.dev/target-namespace"
+	createdByValue     = "TektonHub"
+)
+
+// FinalizeKind removes all resources after deletion of a TektonHub.
+func (r *Reconciler) FinalizeKind(ctx context.Context, original *v1alpha1.TektonHub) pkgreconciler.Event {
+	logger := logging.FromContext(ctx)
+
+	installerSets := original.Status.HubInstallerSet
+	if len(installerSets) == 0 {
+		return nil
+	}
+
+	for _, value := range installerSets {
+		err := r.operatorClientSet.OperatorV1alpha1().TektonInstallerSets().Delete(ctx, value, metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	// List all TektonHub to determine if cluster-scoped resources should be deleted.
+	ths, err := r.operatorClientSet.OperatorV1alpha1().TektonHubs().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list all TektonHubs: %w", err)
+	}
+
+	for _, th := range ths.Items {
+		if th.GetDeletionTimestamp().IsZero() {
+			// Not deleting all TektonHubs. Nothing to do here.
+			return nil
+		}
+	}
+
+	if err := r.extension.Finalize(ctx, original); err != nil {
+		logger.Error("Failed to finalize platform resources", err)
+	}
+	logger.Info("Deleting cluster-scoped resources")
+	manifest, err := r.installed(ctx, original)
+	if err != nil {
+		logger.Error("Unable to fetch installed manifest; no cluster-scoped resources will be finalized", err)
+		return nil
+	}
+	if err := common.Uninstall(ctx, manifest, nil); err != nil {
+		logger.Error("Failed to finalize platform resources", err)
+	}
+	return nil
+}
+
+// ReconcileKind compares the actual state with the desired, and attempts to
+// converge the two.
+func (r *Reconciler) ReconcileKind(ctx context.Context, th *v1alpha1.TektonHub) pkgreconciler.Event {
```

Comment:
first we need the CR name check (as k8s doesn't support singleton CRDs, yet).

Then we need some logic to check if the current loops is an upgrade. 
eg: [here](https://github.com/tektoncd/operator/blob/a0cd67e5c428babf1f76316a14cf9c4526055cab/pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go#L135) without this in upgrades TektonHub reconciler:
1. will not wait for conditions/checks on other resources (even if it can it will be on resources of previous versions)
2. will not reset conditions, without it the conditions will stay true during an upgrade.